src/Api/Endpoints/Projects/Images.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * This implements the Images page class under Project
+ *
+ * PHP Version 7
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris
+ */
+namespace LORIS\Api\Endpoints\Projects;
+
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \LORIS\Api\Endpoint;
+
+/**
+ * A class for handling the /projects/$projectname/images endpoint.
+ *
+ * @category API
+ * @package  Loris
+ * @author   Xavier Lecours Boucher <xavier.lecours@mcin.ca>
+ * @license  Loris license
+ * @link     https://github.com/aces/Loris
+ */
+class Images extends Endpoint implements \LORIS\Middleware\ETagCalculator
+{
+    /**
+     * A cache of the results of the endpoint, so that
+     * it doesn't need to be recalculated for the ETag and handler
+     */
+    protected $responseCache = array();
+
+    /**
+     * Return which methods are supported by this endpoint.
+     *
+     * @return array supported HTTP methods
+     */
+    protected function allowedMethods() : array
+    {
+        return ['GET'];
+    }
+
+    /**
+     * Versions of the LORIS API which are supported by this
+     * endpoint.
+     *
+     * @return array a list of supported API versions.
+     */
+    protected function supportedVersions() : array
+    {
+        return ["v0.0.3-dev"];
+    }
+
+    /**
+     * Handles a request that starts with /projects/$projectname/candidates
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return ResponseInterface The outgoing PSR7 response
+     */
+    public function handle(ServerRequestInterface $request) : ResponseInterface
+    {
+        // FIXME: Validate project based permissions.
+
+        $pathparts = $request->getAttribute('pathparts');
+        if (count($pathparts) > 1) {
+            return (new \LORIS\Http\Response())
+                    ->withStatus(404);
+        }
+
+        return (new \LORIS\Http\Response())
+            ->withHeader("Content-Type", "application/json")
+            ->withBody(
+                new \LORIS\Http\StringStream(
+                    json_encode(
+                        $this->_toArray($request)
+                    )
+                )
+            );
+    }
+
+    /**
+     * Create an array representation of this endpoint's reponse body
+     *
+     * @param ServerRequestInterface $request The incoming PSR7 request
+     *
+     * @return array
+     */
+    private function _toArray(ServerRequestInterface $request): array
+    {
+        $project     = $request->getAttribute('project');
```

Comment:
why not inject this with the constructor instead? (Same with most of the other projects endpoints). It's more of a class attribute than a request attribute and that way you would have better type checking.